pkg/controller/controller.go

```diff
@@ -146,6 +147,15 @@ func NewManagerForRestConfig(conf *config.Config, rc *rest.Config) (ctrl.Manager
 	return m, nil
 }
 
+// IsPrometheusBestPracticeName - function returns true if the name given matches best practices for prometheus name, i.e. snake_case
```

Comment:
Looks pretty good to me. Should be moved to testutils https://github.com/Azure/aks-app-routing-operator/blob/main/pkg/controller/testutils/testutils.go

since this will only be a unit testing check

Reply:
I understand your reasoning. But let me clarify. So it's only a unit testing check at the moment -  only because I like to introduce functionality and its unit test into the code base, before actually using it in throughout the code. But, it will be a function called by every controller before initializing the logger,  so I think its reasonable to put it here. The next PR you will see it called by `cleaner.go`